cmd/utils/flags.go

```diff
@@ -424,12 +426,25 @@ func MakeEthConfig(clientID, version string, ctx *cli.Context) *eth.Config {
 	if err != nil {
 		glog.V(logger.Error).Infoln("WARNING: No etherbase set and no accounts found as default")
 	}
+	// Resolve the mode of opeation from the string flag
+	var clientMode eth.Mode
+	switch strings.ToLower(ctx.GlobalString(EthModeFlag.Name)) {
+	case "archive":
+		clientMode = eth.ArchiveMode
+	case "full":
+		clientMode = eth.FullMode
+	case "light":
+		clientMode = eth.LightMode
+	default:
+		glog.Fatalf("Unknown node type requested: %s", ctx.GlobalString(EthModeFlag.Name))
+	}
```

Comment:
Please drop these. They do not make sense in the context of the ETH protocol any more. The fast sync was already integrated separately as a bool flag. Why do you need to specify this mode to the eth package? If you are a light client, you should simply never even construct an eth object, so I don't see any rational behind modifying the eth.Config and the related things.